.github/workflows/release-publish.yml

```diff
@@ -0,0 +1,151 @@
+        with:
+          fetch-depth: '0'
+          ref: refs/tags/${{ github.event.inputs.releaseTag }}
+
+      # Deploys Maven artifacts. Build and test steps were already ran in previous steps.
+      # Not running tests, because the environment contains secrets.
+      - name: Publish Maven artifacts for release
+        env:
+          # To release with Gradle
+          ORG_GRADLE_PROJECT_signingKey: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
+          ORG_GRADLE_PROJECT_signingPassword: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+          ORG_GRADLE_PROJECT_sonatypeUsername: ${{ secrets.OSSRH_ACCESS_ID }}
+          ORG_GRADLE_PROJECT_sonatypePassword: ${{ secrets.OSSRH_TOKEN }}
+          # To release commits that used Maven to build
+          MAVEN_USERNAME: ${{ secrets.OSSRH_ACCESS_ID }}
+          MAVEN_OSSRH_TOKEN: ${{ secrets.OSSRH_TOKEN }}
+          MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+        run: |
+          echo "::group::Publish to Sonatype"
+          ./gradlew --no-watch-fs --no-daemon publishToMavenLocal publishToSonatype closeAndReleaseSonatypeStagingRepository -Prelease
+          echo "::endgroup::"
+
+          cp cli/build/executable/iceberg-catalog-migrator cli/build/executable/iceberg-catalog-migrator-${RELEASE_VERSION}
+
+          echo "CLI_EXEC=cli/build/executable/iceberg-catalog-migrator-${RELEASE_VERSION}" >> ${GITHUB_ENV}
+
+          echo "## Successfully released ${RELEASE_VERSION} to Sonatype" >> $GITHUB_STEP_SUMMARY
+
+      # Prepare Iceberg Catalog Migrator release notes for GitHub
+      #
+      # The markdown file for the release is generated using some mostly simple shell script.
+      #
+      # `LAST_TAG` is evaluated using `git describe`, which is the name of the git tag before the release tag
+      # `NUM_COMMITS` is the total number of commits "between" LAST_TAG and GIT_TAG
+      #
+      # "Full Changelog" is the output of a `git log` considering the commits "between" LAST_TAG and
+      # GIT_TAG, removing the commits by `renovate` and `nessie-release-workflow`.
+      # Also removes commits that start with `[release] `.
+      #
+      # The final markdown is just a `cat` of the above information including some basic markdown formatting.
+      #
+      - name: Prepare release for GitHub
+        run: |
+          DIR=$(mktemp -d)
+          NOTES_FILE=${DIR}/release-notes.md
+          LAST_TAG=$(git describe --abbrev=0 --tags --match=catalog-migrator-* || echo "catalog-migrator-0.1.0")
+          NUM_COMMITS=$(git log --format='format:%h' ${LAST_TAG}..HEAD^1 | wc -l)
+
+          git log --perl-regexp --author '^(?!.*renovate|.*nessie-release-workflow).*$' --format='format:* %s' ${LAST_TAG}..${GIT_TAG} | grep -v '^\* \[release\] .*$' > ${DIR}/release-log
+
+          Q_CLI_EXEC="https://github.com/projectnessie/iceberg-catalog-migrator/releases/download/iceberg-catalog-migrator-${RELEASE_VERSION}/iceberg-catalog-migrator-${RELEASE_VERSION}"
```

Comment:
I am not sure how the release upload works yet (figuring out)
But `iceberg-catalog-migrator-${RELEASE_VERSION}` is repeated twice at the end. 

Should we make it once as it is identical?